cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -1645,21 +1748,93 @@ Result<FileInfoVector> AzureFileSystem::GetFileInfo(const FileSelector& select)
 
 Status AzureFileSystem::CreateDir(const std::string& path, bool recursive) {
   ARROW_ASSIGN_OR_RAISE(auto location, AzureLocation::FromString(path));
-  if (recursive) {
-    return impl_->CreateDirRecursive(location);
-  } else {
-    return impl_->CreateDir(location);
+  if (location.container.empty()) {
+    return Status::Invalid("CreateDir requires a non-empty path.");
```

Comment:
I purposefully changed it to path because the person reading this error message is passing a path to the filesystem API. The "container" being the first path segment is an internal implementation concern.

Reply:
Makes sense